meson.build

```diff
@@ -51,7 +51,13 @@ cuda_inc_path = get_option('cudapath_inc')
 cuda_lib_path = get_option('cudapath_lib')
 cuda_stub_path = get_option('cudapath_stub')
 
-if cuda_lib_path == ''
+# Check for nvcc compiler availability once at the beginning
+nvcc_prog = find_program('nvcc', required: false)
+
+if not nvcc_prog.found()
+    message('CUDA compiler (nvcc) not found in PATH - disabling CUDA support')
+    cuda_dep = disabler()
```

Comment:
if ncc is not found, it does NOT mean whole cuda support needs to be disabled. only device API tests should be disabled. the rest does not really depend on nvcc.